### DISCUSSION TITLE
Tailwind UI component classes

### DISCUSSION THREAD

**Discussion by jasongitmail** [tailwindlabs/tailwindcss#10897] [General]
I really like [Daisy UI's](https://daisyui.com) approach to component classes, such `.btn`. These classes can be overridden with Tailwind CSS classes when prototyping, or using `@apply` directives like `.btn { @apply text-bold pt-12... }`.

The benefit is that code is DRYer and it reduces changes of component instances differing slightly from how the same component is used elsewhere, unless intentional. Developers can customize this using `@apply` directives to create a custom component system.

Question:
Has Tailwind UI considered using component classes in this manner? If not, are there downsides to this approach?

**Comment by wongjn** [comment]
Tailwind generally advocates avoiding using `@apply` and avoiding constructing your own CSS rules. It is more suited to using your templating language to abstract the components instead of CSS. See https://tailwindcss.com/docs/reusing-styles.

**Comment by jasongitmail** [comment]
That makes sense as a _TailwindCSS_ recommendation, because it's not a component library. 

But for a component library like TailwindUI, it has the downside that if you customize the component styles, it's unclear what styles are from the component library vs your own, making debugging harder.
